src/lib/metricsMgr/MetricsManager.cpp

```diff
@@ -56,84 +94,155 @@ void MetricsManager::accum(const std::string& srv, const std::string& subServ, c
   // Do we have the subservice in the map?
   if (metrics[srv]->find(subServ) == metrics[srv]->end())
   {
+    //
     // not found: create it
     // FIXME PR: this syntax should be simpler, closer to
     // metrics[srv][subServ] = new std::map<std::string, int>;
-
+    //
     metrics[srv]->insert(std::pair<std::string, std::map<std::string, int>*>(subServ, new std::map<std::string, int>));
   }
 
   // Do we have the metric in the map?
   if (metrics[srv]->at(subServ)->find(metric) == metrics[srv]->at(subServ)->end())
   {
+    //
     // not found: create it
     // FIXME PR: I don't like the at() and pair() syntax, I'd prefer a syntax closer to:
     // metrics[srv][subServ][metric] = 0;
+    //
     metrics[srv]->at(subServ)->insert(std::pair<std::string, int>(metric, 0));
   }
 
   metrics[srv]->at(subServ)->at(metric) += value;
+
+  sem_post(&sem);
 }
 
+
+
 /* ****************************************************************************
 *
 * MetricsManager::reset -
 */
 void MetricsManager::reset(void)
 {
+  if (on == false)
+  {
+    return;
+  }
+
+  sem_wait(&sem);
   // FIXME PR (see .h)
+  sem_post(&sem);
 }
 
+
+
 /* ****************************************************************************
 *
 * MetricsManager::toJson -
+*
```

Comment:
Diff is a bit noisy... apart from sem management addition and use variables for the iterators, any change in the "core" logic of the method?